pkg/cloud/services/compute/instance.go

```diff
@@ -142,50 +137,37 @@ func (s *Service) InstanceCreate(openStackCluster *infrav1.OpenStackCluster, mac
 	}
 	input.Networks = &nets
 
-	out, err := createInstance(s, clusterName, input)
+	out, err := createInstance(s, openStackMachine, clusterName, input)
 	if err != nil {
-		record.Warnf(openStackMachine, "FailedCreateServer", "Failed to create server %s: %v", input.Name, err)
 		return nil, err
 	}
-	record.Eventf(openStackMachine, "SuccessfulCreateServer", "Created server %s with id %s", out.Name, out.ID)
 	return out, nil
 }
 
-func createInstance(is *Service, clusterName string, i *infrav1.Instance) (*infrav1.Instance, error) {
-	// Get image ID
-	imageID, err := getImageID(is, i.Image)
-	if err != nil {
-		return nil, fmt.Errorf("create new server err: %v", err)
-	}
-
+func createInstance(is *Service, object runtime.Object, clusterName string, i *infrav1.Instance) (*infrav1.Instance, error) {
 	accessIPv4 := ""
-	networkList := i.Networks
-	portsList := []servers.Network{}
-	for _, network := range *networkList {
+	portList := []servers.Network{}
+
+	for _, network := range *i.Networks {
 		network := network
```

Comment:
If I'm not mssing anything then we could also get rid of this line?

Reply:
Without this, I encountered the following error by `make lint`
```
pkg/cloud/services/compute/instance.go:157:65: G601: Implicit memory aliasing in for loop. (gosec)
                port, err := getOrCreatePort(is, object, clusterName, i.Name, &network, i.SecurityGroups)
```

Reply:
Done